batchservice/src/main/java/nu/yona/server/batch/sample/PinResetConfirmationCodeSenderBatchJob.java

```diff
@@ -0,0 +1,186 @@
+
+@Component
+public class PinResetConfirmationCodeSenderBatchJob
+{
+	private static final int CHUNK_SIZE = 10;
+
+	private static final Logger logger = LoggerFactory.getLogger(PinResetConfirmationCodeSenderBatchJob.class);
+
+	@Autowired
+	private YonaProperties yonaProperties;
+
+	@Autowired
+	private UserService userService;
+
+	@Autowired
+	private PinResetRequestService pinResetRequestService;
+
+	@Autowired
+	private JobRepository jobRepository;
+
+	@Autowired
+	private JobBuilderFactory jobBuilderFactory;
+
+	@Autowired
+	private StepBuilderFactory stepBuilderFactory;
+
+	@PersistenceContext
+	private EntityManager entityManager;
+
+	@Autowired
+	private JpaPagingItemReader<User> reader;
+
+	@Autowired
+	private TaskScheduler scheduler;
+
+	@Bean(destroyMethod = "")
+	@StepScope
+	public JpaPagingItemReader<User> reader(@Value("#{jobParameters[tillDate]}") final Date tillDate)
+	{
+		try
+		{
+			String jpqlQuery = "SELECT u FROM User u WHERE u.pinResetConfirmationCode IS NOT NULL AND u.pinResetConfirmationCode.confirmationCode IS NULL AND u.pinResetConfirmationCode.creationTime < :tillDate";
+
+			JpaPagingItemReader<User> reader = new JpaPagingItemReader<User>();
+			reader.setQueryString(jpqlQuery);
+			reader.setParameterValues(Collections.singletonMap("tillDate", toZonedDateTime(tillDate)));
+			reader.setEntityManagerFactory(entityManager.getEntityManagerFactory());
+			reader.setPageSize(CHUNK_SIZE);
+			reader.afterPropertiesSet();
+			reader.setSaveState(true);
+
+			return reader;
+		}
+		catch (Exception e)
+		{
+			throw YonaException.unexpected(e);
+		}
+	}
+
+	@Bean
+	public ItemProcessor<User, User> processor()
+	{
+		return new ItemProcessor<User, User>() {
+			@Override
+			public User process(final User user) throws Exception
+			{
+				logger.info("Generating pin reset confirmation code for user with mobile number '{}' and ID '{}'",
+						user.getMobileNumber(), user.getID());
+				ConfirmationCode pinResetConfirmationCode = user.getPinResetConfirmationCode();
+				pinResetConfirmationCode.setConfirmationCode(userService.generateConfirmationCode());
+				pinResetRequestService.sendConfirmationCodeTextMessage(user, pinResetConfirmationCode);
+
+				return user;
+			}
+		};
+	}
+
+	@Bean
+	public JpaItemWriter<User> writer()
+	{
+		JpaItemWriter<User> writer = new JpaItemWriter<>();
+		writer.setEntityManagerFactory(entityManager.getEntityManagerFactory());
+
+		return writer;
+	}
+
+	@Bean
+	public Step step1()
+	{
+		return stepBuilderFactory.get("step1").<User, User> chunk(CHUNK_SIZE).reader(reader).processor(processor())
+				.writer(writer()).build();
+	}
+
+	@Bean
+	public Job pinResetConfirmationCodeSenderJob()
+	{
+		return jobBuilderFactory.get("pinResetConfirmationCodeSenderJob").incrementer(new RunIdIncrementer()).flow(step1()).end()
+				.build();
+	}
+
+	@EventListener({ ContextRefreshedEvent.class })
+	void onContextStarted(ContextRefreshedEvent event)
+	{
+		scheduler.scheduleWithFixedDelay(this::runJob,
```

Comment:
Maybe add a comment here that you do this by code as the annotation did not allow expressions.